### ISSUE TITLE
"type(x) is T" not working properly with generic types

### ISSUE THREAD

**Issue by ethframe** [microsoft/pyright#2129]
**Describe the bug**
When `type(x) is T` type guard is used to narrow type of union that contains a generic class, type parameters of that generic class are replaced with `Unknown`.

**To Reproduce**

```python
from typing import Generic, TypeVar, Union,

T = TypeVar('T')


class A(Generic[T]):
    def __init__(self, a: T):
        self.a = a


class B:
    pass


C = Union[A[T], B]


def foo(x: C[T]) -> None:
    if type(x) is A:
        reveal_type(x)
```

Revealed type is `A[Unknown]`.

**Expected behavior**

Revealed type is `A[T@foo]`, as in:

```python
def foo(x: C[T]) -> None:
    if isinstance(x, A):
        reveal_type(x)
```

**VS Code extension or command-line**
Pylance language server 2021.7.5 (pyright 06d75c6c)
pyright 1.1.158

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
